### PR TITLE
Generalised for hospitalised phenotype

### PR DESCRIPTION
**Pull request by rdenholm**
Also includes estout syntax

### CODE REVIEW

**Review by venexia — CHANGES_REQUESTED** [review]

File: analysis/cox_model.do [hunk 1]
```diff
@@ -1,33 +1,96 @@
...
+rename hospitalised_follow_up_end hosp_follow_up_end
+rename non_hospitalised_follow_up_end non_hosp_follow_up_end
+rename hospitalised_censor_date  hosp_censor_date
+rename non_hospitalised_censor_dat non_hosp_censor_date
 
 * Reformat variables
 
-foreach var of varlist exposure_date outcome_date follow_up_start follow_up_end date_expo_censor {
+foreach var of varlist exposure_date outcome_date follow_up_start follow_up_end hosp_follow_up_end non_hosp_follow_up_end hosp_censor_date non_hosp_censor_date {
 	gen `var'_tmp = date(`var', "YMD")
```
[the remark this answers is not in the document]
Comment (rdenholm) [reply analysis/cox_model.do:46]: Done

File: analysis/cox_model.do [hunk 2]
```diff
@@ -40,63 +103,145 @@ replace region_tmp = 8 if region=="West Midlands"
...
+replace covariates_to_fit = subinstr(covariates_to_fit,"ethnicity","cov_cat_ethnicity",1)
+
+replace covariates_to_fit = subinstr(covariates_to_fit,","," ",.) 
+ 
+* replacing collapsed covariates in covariates_to_fit (need to see what they look like) 
+foreach var of varlist cov_cat_ethnicity cov_cat_deprivation cov_cat_smoking_status {
+	replace covariates_to_fit = subinstr(covariates_to_fit,"`var'","`var'_collapsed",1) if substr(covariates_collapsed,1,.)==`var' 
+}
 
+* Creating a list of confounders included in covariates_to_fit, currently does not take into account factor variables
```
[the remark this answers is not in the document]
Comment (rdenholm) [reply analysis/cox_model.do:171]: Updated

File: analysis/cox_model.do [hunk 3]
```diff
@@ -40,63 +103,145 @@ replace region_tmp = 8 if region=="West Midlands"
...
+replace covariates_to_fit = subinstr(covariates_to_fit,"cov_bin_combined_oral_contraceptive_pill","cov_bin_comb_oral_contra",1) 
+replace covariates_to_fit = subinstr(covariates_to_fit,"cov_bin_hormone_replacement_therapy","cov_bin_hormone_replace",1) 
+replace covariates_to_fit = subinstr(covariates_to_fit,"cov_bin_other_arterial_embolism","cov_bin_other_art_embol",1)
+replace covariates_to_fit = subinstr(covariates_to_fit,"cov_bin_chronic_obstructive_pulmonary_disease","cov_bin_copd",1) 
+replace covariates_to_fit = subinstr(covariates_to_fit,"cov_bin_chronic_kidney_disease","cov_bin_ckd",1) 
+replace covariates_to_fit = subinstr(covariates_to_fit,"ethnicity","cov_cat_ethnicity",1)
+
+replace covariates_to_fit = subinstr(covariates_to_fit,","," ",.) 
+ 
+* replacing collapsed covariates in covariates_to_fit (need to see what they look like)
```
[the remark this answers is not in the document]
Comment (rdenholm) [reply analysis/cox_model.do:166]: Removed

File: analysis/cox_model.do [hunk 4]
```diff
@@ -1,33 +1,96 @@
...
+lab val expo_hosp expo_hosp
+
+recode expo_pheno (2=.), gen(expo_non_hosp)
+lab def expo_non_hosp 0"no infection" 1"non hospitalised"
+lab val expo_non_hosp expo_non_hosp
+
+tostring covariates_collapsed, replace
+
+* Reformating data to make dates make more sense if run in pseudo dataset
+do "analysis\pseudo_data.do"
```
[the remark this answers is not in the document]
Comment (rdenholm) [reply analysis/cox_model.do:92]: Removed

File: analysis/cox_model.do [hunk 5]
```diff
@@ -40,63 +103,145 @@ replace region_tmp = 8 if region=="West Midlands"
...
+drop ethnicity
+rename ethnicity_tmp cov_cat_ethnicity
+
+recode cov_cat_ethnicity (2 3 4 5 = 2) (6=3), gen(ethnicity_collapsed)
+lab def ethnicity_collapsed 1 "White" 2 "Non-white" 3 "Missing" 
+lab val ethnicity_collapsed cov_cat_ethnicity_collapsed
+
+gen cov_cat_deprivation_tmp = .
+replace cov_cat_deprivation_tmp = 1 if cov_cat_deprivation=="1-2 (most deprived)"
+replace cov_cat_deprivation_tmp = 2 if cov_cat_deprivation=="03-Apr"
```
[the remark this answers is not in the document]
Comment (rdenholm) [reply analysis/cox_model.do:127]: Updated

**Review by venexia — APPROVED** [review]

**Pull request merged**